src/SFA.DAS.Admin.Aan.Web/Controllers/ManageEvent/SchoolEventController.cs

```diff
@@ -102,12 +121,21 @@ private SchoolNameViewModel GetViewModelEventSchoolName(EventSessionModel sessio
     {
         var searchResult = $"{sessionModel?.SchoolName} (URN: {sessionModel?.Urn})";
 
+        var cancelLink = Url.RouteUrl(RouteNames.NetworkEvents)!;
+
+        if (sessionModel!.HasSeenPreview)
+        {
+            cancelLink = Url.RouteUrl(RouteNames.ManageEvent.CheckYourAnswers)!;
+        }
+
         return new SchoolNameViewModel
         {
             SearchResult = searchResult,
-            CancelLink = Url.RouteUrl(RouteNames.NetworkEvents)!,
+            CancelLink = cancelLink,
             PostLink = Url.RouteUrl(RouteNames.ManageEvent.SchoolName)!,
-            PageTitle = Application.Constants.CreateEvent.PageTitle
+            PageTitle = Application.Constants.CreateEvent.PageTitle,
+            DirectCallFromCheckYourAnswers = sessionModel.DirectCallFromCheckYourAnswers,
```

Comment:
This is bothering me a bit as there is no mention of this in tests anywhere

Reply:
It's in stories CSP-999 and CSP-1001 I think, there is a different behaviour required on the pages they direct to (specifically the 'cancel' is sometimes there, sometimes isn't, depending if you come to a page directly from CheckYourAnswers, or indirectly)